.ci/test_r_package.sh

```diff
@@ -109,10 +124,50 @@ if [[ $TASK == "r-package-check-docs" ]]; then
 fi
 
 cd ${BUILD_DIRECTORY}
-Rscript build_r.R --skip-install || exit -1
 
 PKG_TARBALL="lightgbm_${LGB_VER}.tar.gz"
 LOG_FILE_NAME="lightgbm.Rcheck/00check.log"
+if [[ $R_BUILD_TYPE == "cmake" ]]; then
+    Rscript build_r.R --skip-install || exit -1
+elif [[ $R_BUILD_TYPE == "cran" ]]; then
+
+    # on Linux, we recreate configure in CI to test if
+    # a change in a PR has changed configure.ac
+    if [[ $OS_NAME == "linux" ]]; then
+        cp VERSION.txt R-package/src/VERSION.txt
+        cd ${BUILD_DIRECTORY}/R-package
+        autoconf \
+            --output configure \
+                configure.ac \
+        || exit -1
+        cd ${BUILD_DIRECTORY}
+
+        git diff
```

Comment:
@jameslamb Seems that CI script is fixed now and can catch diffs in configure. Thanks for the fix! Please remove this debug `git diff` and I hope we can finally merge this PR!

Reply:
```suggestion
```

Reply:
ah yep! removed in https://github.com/microsoft/LightGBM/pull/3188/commits/9012752c3d03d8a8e88a11003a14f1e258bde413

Since we have 3 approvals now, I'm planning to merge if all CI jobs pass.